### ISSUE TITLE
In a generator, with a screen reader running, changing the format of a duration can crash Audacity

### ISSUE THREAD

**Issue by DavidBailes** [audacity/audacity#5337]
### Bug description

With a screen reader running, in one of the the built in generators, for example chirp, changing the format of the duration can crash Audacity. More precisely, if the focused digit is in the nth field of the duration, and you change the format to one which doesn't have an nth field, Audacity crashes.

### Steps to reproduce

On Windows, with a screen reader running, eg NVDA or Narrator:
1. Open Audacity.
2. On the Generate menu, choose Chirp.
3. Tab to the duration, and press right arrow to move one digit to the right
4. Open the time control's context menu, and choose seconds.

### Expected behavior

Nothing unexpected.

### Actual behavior

Audacity crashes.

### Audacity Version

latest stable version (from audacityteam.org/download)

### Operating system

Windows

### Additional context

This bug seems to have been introduced in Audacity 3.3.0.

**Comment by chinakov** [comment]
Tested.
